metals/src/main/scala/scala/meta/internal/metals/MetalsLanguageServer.scala

```diff
@@ -1606,6 +1614,17 @@ class MetalsLanguageServer(
         quickConnectToBuildServer().asJavaObject
       case ServerCommands.DisconnectBuildServer() =>
         disconnectOldBuildServer().asJavaObject
+      case ServerCommands.DecodeFile() =>
+        val argsMaybe = Option(params.getArguments())
+        argsMaybe.flatMap(_.asScala.headOption) match {
+          case Some(arg: JsonPrimitive) =>
+            val uri = arg.getAsString()
+            fileDecoderProvider.decodedFileContents(uri).asJavaObject
+          case _ =>
+            val error = s"Invalid DecodeFile arguments $params"
+            scribe.error(error)
+            Future { error }.asJavaObject
```

Comment:
I'd recommend handling this a little different. Since no matter what this seems to just be returning a string, for a client there won't be a good way to determine if they are getting back a valid string, or just this string which is really an error. Maybe instead of just returning the error string here we could show a message to the user letting them know this and don't return the error string as a valid response like we are doing now.